### ISSUE TITLE
Incorrect `UseComboBoxState` Types

### ISSUE THREAD

**Issue by kltdwrds** [downshift-js/downshift#968]
- `downshift` version: `^5.0.4`
- `node` version: `v12.10.0`
- `npm` (or `yarn`) version: `6.13.7`

**Relevant code or config**

```javascript
    const {
      selectedItem
    } = useCombobox({
      items: filteredGuides,
      initialSelectedItem: undefined
 });

   selectedItem.id // error, selectedItem is undefined
```

**Problem description**: 
`UseComboBoxState` TypeScript type is incorrect, `selectedItem` and `highlightedItem` can be undefined

Current:

https://github.com/downshift-js/downshift/blob/f6a7913f6cb542864214373c4ad5fb5cc74f00d6/typings/index.d.ts#L373-L378

**Suggested solution**:
Make the attribute optional:
``` 
export interface UseComboboxState<Item> {
  highlightedIndex?: number
  selectedItem?: Item
  isOpen: boolean
  inputValue: string
}
```

**Comment by silviuavram** [comment]
In what situation they end up as undefined (apart from controlling them yourself) ? For instance we start with them as `-1` and `null` (in initial state). So it will be helpful to pinpoint to the scenario.

**Comment by kltdwrds** [comment]
If you pass undefined to `initialSelectedItem` then `selectedItem` will be
undefined.

If `selectedItem` is initialized as null, then the type should reflect that
too - I'll update the PR.

**Comment by kltdwrds** [comment]
I removed highlightedIndex from the issue description, it was an oversight in my original post.

**Comment by kltdwrds** [comment]
Also, it's surprising if the `UseComboBoxState` attribute types differ from their corresponding UseComboBoxParams type

**Comment by silviuavram** [comment]
I will take a closer look at this. Thanks for updating the PR!

**Comment by silviuaavram** [comment]
I am going to resume going through the issues this week, I was busy with `useMultipleSelection`, sorry about it. Will definitely handle this!

**Comment by silviuaavram** [comment]
I still don't agree with these changes. You are passing `undefined` for props that are meant to control the hooks' own state. That's should not be allowed, you either want to control the state or not.

I added this https://github.com/downshift-js/downshift/issues/974 into the backlog specifically to add to the hooks the same warnings from moving from controlled to uncontrolled state and back. Indeed, it's related to the full control of state via props but in your PR you are proposing to have the values in the state optional.

To recap, if you pass `undefined` to `selectedItem` or `highlightedIndex` you should not be allowed because these are control props. They don't take `undefined`.
If you pass `undefined` to `initial` / `default` then I really don't see the point in doing that. It does make no sense to me, those are just to help you have initial values in case you don't want to start with our initial values.
